### PR TITLE
Zahin/pm 15

### PR DESCRIPTION
**Pull request by zahin-mohammad**

### CODE REVIEW

**Comment by zahin-mohammad** [comment]
Changed my code to match @brian-norman 's. Will add graph in separate pr.

File: app/src/main/java/io/mochahub/powermeter/stats/StatsFragment.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import kotlinx.android.synthetic.main.fragment_stats.*
+
+
+
+
+
+class StatsFragment : Fragment() {
+
+    private lateinit var statsAdapter : StatsAdapter
+    private lateinit var viewModel: StatsViewModel
```
Comment (brian-norman) [inline app/src/main/java/io/mochahub/powermeter/stats/StatsFragment.kt]:
```suggestion
    private val viewModel: StatsViewModel by viewModels()
```

File: app/src/main/java/io/mochahub/powermeter/stats/StatsFragment.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        super.onCreateView(inflater, container, savedInstanceState)
+        return inflater.inflate(R.layout.fragment_stats, container, false)
+    }
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+        viewModel = ViewModelProviders.of(this).get(StatsViewModel::class.java)
```
Comment (brian-norman) [inline app/src/main/java/io/mochahub/powermeter/stats/StatsFragment.kt]:
```suggestion
```

File: app/src/main/res/layout/fragment_stats.xml [hunk 3]
```diff
@@ -1,19 +1,20 @@
...
+        android:layout_height="match_parent"
+        android:orientation="vertical"
+        tools:context=".stats.StatsFragment">
+
+<!--    <Switch-->
+<!--            android:id="@+id/switch1"-->
+<!--            android:layout_width="wrap_content"-->
+<!--            android:layout_height="wrap_content"-->
+<!--            android:padding="16dp"-->
+<!--            />-->
```
Comment (brian-norman) [inline app/src/main/res/layout/fragment_stats.xml]: 😱

File: app/src/main/res/drawable/card_background.xml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+    <item>
+        <shape android:shape="rectangle">
+            <!-- put your background color here-->
+            <corners android:radius="15dp"/>
+            <solid android:color="@color/design_default_color_background" />
+        </shape>
+    </item>
+
+</ripple>
```
Comment (brian-norman) [inline app/src/main/res/drawable/card_background.xml]: Where is this being used and why ?

File: app/src/main/res/layout/row_stat.xml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:layout_marginBottom="8dp"
+                    android:layout_marginHorizontal="16dp"
+                    android:textColor="#555" />
+        </LinearLayout>
+        <LinearLayout
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content">
+    </LinearLayout>
```
Comment (brian-norman) [inline app/src/main/res/layout/row_stat.xml]: What is this guy doing here!

File: app/src/main/res/layout/row_stat.xml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+                    android:id="@+id/card_description"
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:layout_marginBottom="8dp"
+                    android:layout_marginHorizontal="16dp"
+                    android:textColor="#555" />
+        </LinearLayout>
+        <LinearLayout
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content">
```
Comment (brian-norman) [inline app/src/main/res/layout/row_stat.xml]: Who's mans is this

**Review by brian-norman — CHANGES_REQUESTED** [review]
Some questions and suggestions 😄

File: app/src/main/res/layout/row_stat.xml [hunk 7]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.cardview.widget.CardView xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (brian-norman) [inline app/src/main/res/layout/row_stat.xml]:
```suggestion
com.google.android.material.card.MaterialCardView xmlns:android="http://schemas.android.com/apk/res/android"
```

File: app/src/main/res/layout/row_stat.xml [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+                    android:layout_height="wrap_content"
+                    android:layout_marginBottom="8dp"
+                    android:layout_marginHorizontal="16dp"
+                    android:textColor="#555" />
+        </LinearLayout>
+        <LinearLayout
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content">
+    </LinearLayout>
+</androidx.cardview.widget.CardView>
```
Comment (brian-norman) [inline app/src/main/res/layout/row_stat.xml]:
```suggestion
</com.google.android.material.card.MaterialCardView>

```

File: app/src/main/java/io/mochahub/powermeter/stats/StatsFragment.kt [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+import androidx.recyclerview.widget.LinearLayoutManager
+import io.mochahub.powermeter.R
+import kotlinx.android.synthetic.main.fragment_stats.*
+
+
+
+
+
```
Comment (brian-norman) [inline app/src/main/java/io/mochahub/powermeter/stats/StatsFragment.kt]: 😨

File: app/src/main/res/drawable/card_background.xml (hunk 4 above)
Comment (zahin-mohammad) [reply app/src/main/res/drawable/card_background.xml]: No longer needed.

**Review by brian-norman — APPROVED** [review]
L G T M

**Pull request merged**